beard/metrics/string_distance.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2014, 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""String distances metrics.
+
+.. codeauthor:: Petros Ioannidis <petros.ioannidis91@gmail.com>
+
+"""
+def substring_occurences(s, letter):
+    """Return the number of matching letters based on the jaro algorithm.
+
+    Parameters
+    ----------
+    :param s: string
+        string to be searched
+
+    :param letter: string
+        substring we are searching for
+
+    Returns
+    -------
+    :returns: generator
+        a generator that holds the value of the index
+    """
+    if letter == '':
+        raise ValueError('Empty string is not allowed as a parameter for letter')
+    try:
+        index = s.find(letter)
+    except AttributeError:
+        raise TypeError('S must be a string')
+
+    while index != -1:
+        yield index
+        index = s.find(letter, index + 1)
```

Comment:
You are right. I runned timeit and the code you proposed runs faster. I will change it.